### DISCUSSION TITLE
Bypass Login?

### DISCUSSION THREAD

**Discussion by sup3rm3ga** [koush/scrypted#250] [Q&A] [answered]
Is it possible to bypass login page? Ultimately it would be really great to allow read-only access to a Dashboard somehow for a variety of use cases.

**Comment by linedpaper** [comment]
I am looking for this functionality as well, but not for read-only.  I use Authelia for authentication on my Traefik instance, it would be nice to disable authentication so I can use the existing sso setup I have.  This would apply for anyone using any sort of auth forwarding on a reverse proxy.

**Comment by koush** [comment, marked as answer]
You can add basic authentication headers using a reverse proxy. disabling auth will not be implemented.

**Reply by skjdghsdjgsdj** [reply]
Can you please elaborate on this? What values would I use for the username and password; the same ones as an internal Scrypted user?

I also saw you mentioned using a `SCRYPTED_ADMIN_TOKEN`. Is that something I send as an HTTP header from the SSO proxy?

**Reply by v-oleksandr** [reply]
Hey! 
@skjdghsdjgsdj did you manage to make it work? I'm trying to send basic auth header: 
```
Authorization: Basic base64encoded(scrypteed_username:password)
```
but seems like the header is ignored. 

as others guys here, I would like to use my own SSO for scrypted (authentik in my case)

**Reply by skjdghsdjgsdj** [reply]
I tinkered briefly but couldn't figure it out (yet). I'm still interested in a solution if anybody has ideas.

**edit**: to be clear, I'm using Authentik also.

**Reply by v-oleksandr** [reply]
I see in the code base that it should try to parse basic auth header: https://github.com/koush/scrypted/blob/b684ced6298f4bcdc9bbd3dcb6ceb6d413e04e2d/packages/auth-fetch/src/auth-fetch.ts#L59 
but no luck in my case.. 
@koush any suggestions here? I've also tried to send header via Postman to exlude wrong setup of my authentik + reverse proxy.

**Reply by v-oleksandr** [reply]
@skjdghsdjgsdj https://github.com/koush/scrypted/issues/1750

**Reply by skjdghsdjgsdj** [reply]
Nifty. I'll try that when I get a moment.

**Reply by skjdghsdjgsdj** [reply]
That works!

The tl;dr for Authentik, with an assumption of basic Authentik and reverse proxy usage:

1. Set up a new application and proxy provider with your desired external hostname and an internal hostname pointing to the Scrypted instance.
2. Bind the provider to your applicable, and likely default, provider.
3. In the config for Scrypted--the Docker compose yml in my case--add a key `SCRYPTED_DEFAULT_AUTHENTICATION` and the value of an existing username in Scrypted.
4. Connect to the external hostname you defined in step 1. If you're using a reverse proxy, configure such a host and have it use Authentik as an origin.

After all this, and restarting Scrypted and any other touched services, I now get auto-logged-into Scrypted when I launch it from Authentik's apps. I get logged in as me, not necessarily the same username as the Authentik user. That would be the main drawback, although it's not an issue for my use case.
